docs/source/concept_guides/quantization.mdx

```diff
@@ -1,5 +1,251 @@
+
+
+#### Number of quantization parameters
+
+Depending on the accuracy / latency trade-off you are targetting you can play with the granularity of the quantization parameters:
+
+- Quantization parameters can be computed on a *per-tensor* basis, meaning that one pair of `(S, Z)` will be used per
+tensor.
+- Quantization parameters can be computed on a *per-channel* basis, meaning that it is possible to store a pair of
+`(S, Z)` per element along one of the dimensions of a tensor. For example for a tensor of shape `[B, N, H, W]`, having
+*per-channel* quantization parameters for the second dimension would result in having `N` pairs of `(S, Z)`. While this
+can give a better accuracy, it requires more memory.
+
+#### Calibration
+
+The section above described how quantization from `float32` to `int8` works, but one question
+remains: how is the `[a, b]` range of `float32` values determined? That is where calibration comes in to play.
+
+Calibration is the step during quantization where the `float32` ranges are computed. For weights it is quite easy since
+the actual range is know at *quantization-time*. But it is less clear for activations, and different approaches exist:
+
+1. Post training dynamic quantization: the range for each activation is computed at *runtime*. While this gives great
+results without too much work, it can be a bit slower than static quantization because the range is computed each time.
+It is also not an option on certain hardware.
+2. Post training static quantization: the range for each activation is computed at *quantization-time*.
+  - Observers are put on activations to record their values.
+  - A certain number of forward passes on a calibration dataset is done (around `200` examples is enough).
+  - The ranges for each computation are computed according to some *calibration technique*.
+3. Quantization aware training: the range for each activation is computed at *training-time*, following the same idea
+than post training static quantization. But "fake quantize" operators are used instead of observers: they record
+values just as observers do, but they also simulate the error induced by quantization to let the model adapt on it.
+
+
+For both post training static quantization and quantization aware training, it is necessary to define calibration
+techniques, the most common are:
+
+- Min-max: the computed range is `[min observed value, max observed value]`, this works well with weights.
+- Moving average min-max: the computed range is `[moving average min observed value, moving average max observed value]`,
+thiw works well with activations.
+- Histogram: records a histogram of values along with min and max values, then chooses according to some criterion.
+  - Entropy: the range is computed as the one minimizing the error between the full-precision and the quantized data.
+  - Mean Square Error: the range is computed as the one minimizing the mean square error between the full-precision and
+  the quantized data.
```

Comment:
Add percentile maybe?

Reply:
Done, can you validate that it is correct please?

Reply:
Actually, according to what I see [here](https://github.com/microsoft/onnxruntime/blob/2cb12caf9317f1ded37f6db125cb03ba99320c40/onnxruntime/python/tools/quantization/calibrate.py#L706) In ORT's source code, it seems that it's rather `[p/2% of the observed values are bigger than this one, p/2% of the observed value are smaller than this one]`.

Reply:
It depends on whether the scheme is symmetric or not, from what I could understand:

- When symmetric, only one value can be chosen since the range needs to be `[-a, a]` in the end. So `a` is the value so that p% of the people are smaller.
- When asymmetric, the idea is to have a range when p% of the people fall into, so we cut both ends by the p/2% of the people that do not fall.

Reply:
Yeah the symmetric case is a bit different but I didn't want to make things more complicated :grin: 
Yep I understand it the same way as you do. Your explanation sounds good to me, no need to make the documentation too complicated here :+1: